githooks/check-commit.sh

```diff
@@ -0,0 +1,236 @@
+        echo "::error title=ERROR::${1}"
+    fi
+    FAILED=1
+}
+
+function Warning {
+    if [ -z "$NONINTERACTIVE" ]; then
+        echo "WARNING: ${1}"
+    else
+        echo "::warning title=WARNING::${1}"
+    fi
+    WARNED=1
+}
+
+# Checks count of given field
+function Check_count {
+    local field
+    local count
+    local dest
+    local allowmany
+
+    field="$1"
+    dest="$2"
+    allowmany="${3:-0}"
+
+    count="$(grep -c -e "^${field}:" "$dest" || true)"
+
+    case "$count" in
+    0)
+        Error "Missing ${field} field"
+    ;;
+    1)
+        : # Ok
+    ;;
+    *)
+        if [ "$allowmany" == "0" ]; then
+            Error "Multiple ${field} fields (Only one required and allowed)"
+        fi
+    ;;
+    esac
+}
+
+function Exit_message {
+    if [ -z "$NONINTERACTIVE" ]; then
+        echo ""
+        echo "Your commit message is not lost (yet), it's saved in the .git dir of the repo"
+        echo "You probably can use something like this to edit your message:"
+        echo "git commit -e --file=\$(git rev-parse --git-dir)/COMMIT_EDITMSG"
+        echo ""
+    fi
+    exit 2
+}
+
+set -e
+
+MATCH=1
+while [ "$MATCH" -eq 1 ]; do
+    case "${1,,}" in
+    check-script)
+        if shellcheck "$0" && bashate -i E006 "$0"; then
+            echo "Nothing to complain"
+        fi
+        exit 1
+    ;;
+    ""|help|-h|--help)
+        echo "This script is supposed to be called from the git commit-msg hook (or check-commits.sh)"
+        echo ""
+        echo "check-commit.sh [--noninteractive] COMMIT_MSG_FILE      Check commit message (noninteractively if specified)"
+        echo "check-commit.sh check-script                            Run shellcheck and bashate on the script itself"
+        echo "check-commit.sh [help|-h|--help]                        Show this help"
+        echo ""
+        exit 1
+    ;;
+    --noninteractive)
+        NONINTERACTIVE=1
+        shift
+    ;;
+    *)
+        MATCH=0
+    ;;
+    esac
+done
+
+DEST="$1"
+
+if [ -z "$NONINTERACTIVE" ]; then
+    # Remove trailing spaces
+    sed -i 's/[[:space:]]*$//g' "$DEST"
+
+    # Remove preceding spaces from subject line
+    sed -i '1 s/^[[:space:]]*//' "$DEST"
+
+    # Find 'Jira-Id:' field case insensitively and replace with exact string 'Jira-Id: '
+    # Allow dashes, spaces, underscores and nothing between 'Jira' and 'Id'
+    # Remove extra spaces.
+    #sed -i 's/^[[:blank:]]*[jJ][iI][rR][aA][ _-]*[iI][dD][[:blank:]]*:[[:blank:]]*/Jira-Id: /g' "$DEST"
+
+    # Reformat Signed-off-by field
+    sed -i 's/^[[:blank:]]*[sS][iI][gG][nN][eE][dD][ _-]*[oO][fF][fF][ _-]*[bB][yY][[:blank:]]*:[[:blank:]]*/Signed-off-by: /g' "$DEST"
+
+    # Reformat Change-Id field
+    #sed -i 's/^[[:blank:]]*[cC][hH][aA][nN][gG][eE][ _-]*[iI][dD][[:blank:]]*:[[:blank:]]*/Change-Id: /g' "$DEST"
+
+    # Reformat Depends-On fields
+    #sed -i 's/^[[:blank:]]*[dD][eE][pP][eE][nN][dD][sS][ _-]*[oO][nN][[:blank:]]*:[[:blank:]]*/Depends-On: /g' "$DEST"
```

Comment:
Why do you want to keep the commented-out sections?
This same issue repeats elsewhere in this file too.

Reply:
Originally they were left in there, in case those fields are ever needed again.
In this particular case they were there, because I copied these scripts as they are in our other related repos.
However, commented-out code removed.